### DISCUSSION TITLE
openDTU loses connection to Hoymiles very often

### DISCUSSION THREAD

**Discussion by Nobster18** [tbnobody/OpenDTU#1334] [General]
Hi guys,
I have a serious problem with my openDTU and to be honest I don't know what to do. Maybe someone has an idea ...

I have wired an ESP32 with an antenna from AZDelivery "on the fly", attached a display, configured a little, printed a case with my 3D printer - and the world is colourful. So far, so good ...

I have had the DTU in operation for a few days now and it loses the connection to the Hoymiles 1500 inverter, sometimes up to several times a day, even though the DTU is not moving :-). Sometimes, however, it works without problems for a whole day. Since I tap the data for a long-term display via a RaspberryPi with "Solaranzeiger", I get incorrect values again and again.

Unfortunately, restarting the DTU via the web interface is not enough to get the connection back, but I have to disconnect the DTU from the power supply. The transmitting power of the NRF24 antenna is already set to maximum, so I can't do any more. Between the DTU and the inverter there are a few roof tiles, insulating foil and a wooden ceiling boarded up with quick construction boards. No reinforced concrete or similar...

Unfortunately, I'm running out of ideas - do you have any more?

Thanks and greetings,
Norbert

**Comment by huste511** [comment]
Info -> Console:
Which reception level (-xx dBm) is displayed?

btw: "foil" as today's vapor barrier is the plague for RF. ;-(

**Reply by Nobster18** [reply]
Hi,

there is ..... -80 dBm

**Reply by huste511** [reply]
And there you already have your answer.
-80 is not very good.
Try a different antenna orientation or place. If necessary, buy a patch antenna and omnidirectional antenna with more amplification, reduce the transmission power accordingly.

**Reply by Nobster18** [reply]
What should I try to achieve?

**Reply by huste511** [reply]
Values ​​that are higher, i.e. in the direction of -70 ... -60 ...
nrf24l01+ has -94dBm sensitivity but as you can see, it's not always enough.
Your -80 were only an indication if the connection is established.

Compare it with your values ​​for WiFi Info -> Network -> RSSI

What kind of external antenna do you have exactly?
Or is this on the circuit board?

**Reply by Nobster18** [reply]
RSSI is at 82.
I tried another place in the house - 3 meters away, but -80 dBm is still unchanged.

A few months ago I bought a nRF24L01 - you can take a look here:
https://www.amazon.de/gp/product/B07PQPFTWZ/ref=ppx_yo_dt_b_asin_title_o07_s00?ie=UTF8&psc=1

**Reply by huste511** [reply]
nRF24L01 or nRF24L01+?
nRF24L01 has only -82/-85dBm sensitivity

**Reply by Nobster18** [reply]
nRF24L01 - if the description at Amazon is correct

**Reply by Nobster18** [reply]
I changed the settings for the DTU at the website from minimum -18 dBm up to maximum 0 dBm. But regardless what I choose in the console I got -80 dBm.
Will try the setup with the electrolytic capacitor in the next 2 weeks - because I will be in holiday from Saturday on ...

**Reply by huste511** [reply]
The information in the console is from the received signal.
You adjust the transmission power.

The polarization of the antennas also makes a difference. With the MWR they are usually horizontal.
The antenna at the base should therefore ideally be horizontal and present the MWR with as much side surface as possible.

However, within sight of the MWR without wiping metal, the values ​​should increase significantly.
A year ago I read about defective/bad modules.

**Comment by Solarteur** [comment]
- The maximum transmission power for the nRF module is often **not** helpful. Have you tested other settings?
- Is an approx. 100µF electrolytic capacitor soldered between 3.3V and GND of the nRF module?
- Have you tested a different power supply/cable?

**Reply by Nobster18** [reply]
1. Yes, all other settings were tested - but not better. I started with the standard settings ...
2. No, actually not. I just ordered a board and a capacitor, because I thought that could be a possible solution
3. Yes, many power supplies were tested. Currently its a Xiaomi power supply with 90 Watt. But there was no difference

I also tried some other places in the building to use the openDTU, but it wasn't better.
